velox/exec/tests/TableScanTest.cpp

```diff
@@ -2467,3 +2468,35 @@ TEST_F(TableScanTest, parallelPrepare) {
   }
   AssertQueryBuilder(plan).splits(splits).copyResults(pool_.get());
 }
+
+TEST_F(TableScanTest, scanPartitionKey) {
+  parquet::registerParquetReaderFactory(parquet::ParquetReaderType::NATIVE);
+  // bigint.parquet holds one columns (id:BIGINT) and 1 row [0].
+  auto filePath = facebook::velox::test::getDataFilePath(
+      "velox/exec/tests", "data/bigint.parquet");
+  auto split = HiveConnectorSplitBuilder(filePath)
+                   .fileFormat(dwio::common::FileFormat::PARQUET)
+                   .start(0)
+                   .length(472)
+                   .partitionKey("k", "0")
+                   .build();
+  auto outputType = ROW({"n1_0", "n1_1"}, {BIGINT(), BIGINT()});
+  auto subfieldFilters = singleSubfieldFilter("k", lessThanOrEqual(2));
```

Comment:
Does this issue happen only when we have a subfield-filter on a partition key? If that's the case, then it is a bug in the query planner. Such filters should not be pushed down into table scan, but rather processed during split enumeration. We can add a check to TableScanNode constructor to reject such filters.

Reply:
Yes, this is the case, a subfield-filter on a partition key. Ok I can add a check at TableScanNode  constructor.

Reply:
@zhejiangxiaomai This is probably a valid use case. If so, then we should keep these filters separately, not include these in ScanSpec, and evaluate these without creating any file reader by just looking at the partition key values in the split.

Reply:
Here is a simple case in spark sql UT. https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala#L1332-L1359

Reply:
@zhejiangxiaomai Thanks. This looks like a join on partition key like the example above.

Reply:
> Does this issue happen only when we have a subfield-filter on a partition key? If that's the case, then it is a bug in the query planner. Such filters should not be pushed down into table scan, but rather processed during split enumeration. We can add a check to TableScanNode constructor to reject such filters.

Hi @mbasmanova, I believe there are two levels of filter in TableScan, the subfield-filter and the remaining-filter. And we have implemented certain logic to separated the filter conditions in TableScan into these two parts. In above case, the filter condition is created as subfield-filter. Should we change it to remaining-filter or prevent it in TableScan?

Reply:
@rui-mo It would be more efficient to handle subfield filters on a partition keys in the split generation logic by dropping the splits.

Reply:
Thank you, I see. Maybe we need to follow the code here to skip the split if it does not satisfy the filter. Is that correct?

https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L447-L452

cc @@zhejiangxiaomai